### DISCUSSION TITLE
Remove `truelearn.utils.persistent`

### DISCUSSION THREAD

**Discussion by yuxqiu** [TrueLearnAI/truelearn#36] [General]
Currently, we keep the `truelearn.utils.persistent` in our package. However, it seems that they are redundant since we have already supported the method to convert `AbstractKnowledgeComponent` to `dict`. Given the `dict`, it's possible for users to use their library to convert the dictionary to other formats.

Thus, I propose that we could probably remove `truelearn.utils.persistent`.

**Comment by yuxqiu** [comment]
Could Have Reasons:
- probably could provide customization if we encapsulate the `dump/load` methods

If we have time:
- Implement JSON/pickle dump
- Implement JSON/pickle load

**Reply by KD-7** [reply]
Yea i agree with this, we can have a to_json and load_json functions etc

**Comment by KD-7** [comment]
After considering this option we decided to remove persistant, in future something similar to this could be added that made use of [PMML](https://dmg.org/pmml/v4-4-1/GeneralStructure.html). The justification being we generally wanted to follow the scikit-learn convention, and wrapping one line commands to pickle/json dump in this module wouldnt have followed this or been significantly useful.
